scene/3d/ray_cast_3d.cpp

```diff
@@ -185,7 +192,14 @@ void RayCast3D::_notification(int p_what) {
 			if (prev_collision_state != collided && get_tree()->is_debugging_collisions_hint()) {
 				if (debug_material.is_valid()) {
 					Ref<StandardMaterial3D> line_material = static_cast<Ref<StandardMaterial3D>>(debug_material);
-					line_material->set_albedo(collided ? Color(1.0, 0, 0) : Color(1.0, 0.8, 0.6));
+
+					Color collide_color;
+					if (debug_shape_color.r > 0.75 && debug_shape_color.g + debug_shape_color.b < 0.25) {
+						collide_color = (debug_shape_color + 2.0 * Color(0.0, 1.0, 0.0)) / 3.0;
+					} else {
+						collide_color = (debug_shape_color + 2.0 * Color(1.0, 0, 0)) / 3.0;
+					}
```

Comment:
It's a bit difficult to understand the rationale behind this color selection logic. Can you add some comments?

Reply:
Comments added.

Explanation : If the base color is already quite reddish, the base color is mixed to get a greener color.
Else, the base color is highligthed by adding a red component. 

So the highlighted color is always different enough from the custom base color and can be seen easily.

Thanks for your review and comments.

Reply:
Finally, I've modified the color detection to use HSV components. Reddish color are detected more accuratly.
The highligh color isn't a mix any more. It's either red or green (for red raycast). It's more clear that it's a colliding color.
Alpha of the custom color is preserved in the highlighted color.